README.md

```diff
@@ -247,6 +248,23 @@ and add to your init.vim:
 nnoremap z= <Cmd>call VSCodeNotify('keyboard-quickfix.openQuickFix')<CR>
 ```
 
+### Smart Line Numbers
+
+Some popular plugins like [vim-numbertoggle](https://github.com/jeffkreeftmeijer/vim-numbertoggle) or
+[numbers.vim](https://github.com/myusuf3/numbers.vim/) add a feature where relative line numbers are used in normal
+mode, but absolute numbers are used in insert mode.
+
+To replicate this functionality, first make relative numbers default by adding `"editor.lineNumbers": "relative"` to
+your vscode `settings.json`. Then, add the following to your `init.vim` to automatically switch between the two modes:
+
+```vim
+augroup smartnnumbers
+  autocmd!
+  autocmd BufEnter,InsertLeave * RelativeNumber
+  autocmd BufLeave,InsertEnter * Number
+augroup END
```

Comment:
Instead of adding custom commands, vscode-neovim can listen to the `OptionSet` autocmd. However we probably don't want to actually have sign/fold/relnum columns actually enabled in the nvim server, so we would want to do something like:

```
augroup vscode-neovim
  autocmd!
  autocmd OptionSet * call VscodeNvimHandleOptionSet(deepcopy(v:event))
augroup END

func! VscodeNvimHandleOptionSet(ev) abort
    " set the relevant vscode settings...
    ...

    " then UNSET the nvim-side settings, because the sign/fold/relnum columns aren't useful to us.
    " use `noau` to ensure that OptionSet isn't circularly triggered.
    noau set nonumber
    noau set norelativenumber
    noau set signcolumn=no
    noau set foldcolumn=0
endfunc
```